src/watertap_contrib/seto/costing/units/chemical_softening_0D.py

```diff
@@ -0,0 +1,833 @@
+            domain=NonNegativeReals,
+            units=blk.costing_package.base_currency,
+        )
+
+    blk.lime_sludge_mngt_op_cost = Var(
+        initialize=1e5,
+        doc="Operational cost of lime sludge management",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.admin_op_cost = Var(
+        initialize=1e5,
+        doc="Operational cost of administrative, laboratory and maintenance building",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    # endregion
+
+    # region Capital cost component constraints
+
+    # Mixing tank
+    if value(blk.unit_model.vel_gradient_mix) <= 300:
+        blk.mix_tank_capital_cost_constraint = Constraint(
+            expr=blk.mix_tank_capital_cost
+            == (
+                0.0002
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 2
+            )
+            + (
+                22.776
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                + 28584
+            )
+        )
+
+    elif (
+        value(blk.unit_model.vel_gradient_mix) > 300
+        and value(blk.unit_model.vel_gradient_mix) <= 600
+    ):
+        blk.mix_tank_capital_cost_constraint = Constraint(
+            expr=blk.mix_tank_capital_cost
+            == (
+                0.0002
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 2
+            )
+            + (
+                29.209
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                + 30388
+            )
+        )
+
+    elif (
+        value(blk.unit_model.vel_gradient_mix) > 600
+        and value(blk.unit_model.vel_gradient_mix) <= 1000
+    ):
+        blk.mix_tank_capital_cost_constraint = Constraint(
+            expr=blk.mix_tank_capital_cost
+            == (
+                0.0002
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 2
+            )
+            + (
+                55.443
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                + 29756
+            )
+        )
+
+    else:
+        print("Error - Out of bounds (300 - 1000 s-1)")
```

Comment:
It shouldn't be possible to get to this point. It could be a configuration check or just properly bounding the velocity gradient variable. But this goes to my earlier comment on we don't want to be using value in build functions. 

If it needs to be an error, we want to ``raise ValueError`` or something like that so that code execution stops, rather than just a print statement. Because if code execution continues ``mix_tank_capital_cost`` ends up being unconstrained.

Reply:
I agree.